Upgrade to 12.10.14, Step 1 of the upgrade path to 15.6.X

skumar-jnj opened the pull request:
JATV-631

Part of the upgrade path outlined [here](https://docs.gitlab.com/ee/update/#upgrade-paths).  v12.10.14 is the last 12.X release before 13.0, which will require a postgres version update.

Builds Successfully as of 1/23/23

Notes (also in code):
```
# Added two install packages: shared-mime-info libsqlite3-dev
#
# 1) shared-mime-info
# ----------------------
#   In 2021, mimemagic v0.3.6, a dependency of rails, switched its licensing to GPL, 
#   breaking the Gitlab build for versions < v14.00-ee.  Note that a "bundle update mimemagic"
#   is also required in the install.sh script.
#   The issue is described here:  https://github.com/rails/rails/issues/41750
#   The workaround is described here: https://github.com/mimemagicrb/mimemagic#dependencies
#
# 2) libsqlite3-dev
# -------------------
#   The following error was encountered when building the image and 
#   adding libsqlite3-dev to the install appeared to fix things:
#
# Gem::Ext::BuildError: ERROR: Failed to build gem native extension.
#     current directory: /var/lib/gems/2.6.0/gems/sqlite3-1.3.13/ext/sqlite3
# /usr/bin/ruby2.6 -I /usr/lib/ruby/2.6.0 -r ./siteconf20230123-22221-owir04.rb
# extconf.rb
# checking for sqlite3.h... no
# sqlite3.h is missing. Try 'brew install sqlite3',
# 'yum install sqlite-devel' or 'apt-get install libsqlite3-dev'
# and check your shared library search path (the
# location where your sqlite3 shared library is located).
# *** extconf.rb failed ***
```

Commit:
Accelerate map_uidgid

Hi,

I use the function "map_uidgid" and each restart of the container is at least 30 seconds for the command chown.
With this change I am 1 second.

I added the path ${GITLAB_HOME} at the end to not get an error if the "find" finds nothing.

does this change suit you?

eiabea committed:
Fixed readme to match new backup file naming

eiabea committed:
Retrigger travis build

eiabea committed:
Merge branch 'master' into master

eiabea committed:
Merge branch 'master' into master

eiabea committed:
Merge branch 'master' into master

emon committed:
fix gitlab_configure_oauth_saml_attribute_statements()

OAUTH_SAML_ATTRIBUTE_STATEMENTS_USERNAME reflects username: in gitlab.yml
OAUTH_SAML_ATTRIBUTE_STATEMENTS_NAME is not required.

emon committed:
Merge remote-tracking branch 'upstream/master' into oauth_saml_attribute_statements

solidnerd committed:
Fix HostKey function for ubuntu:bionic

solidnerd committed:
Upgrade GitLab Image to 12.2.1-1

solidnerd committed:
release 12.2.1-1

sue445 committed:
Add Sentry configuration to gitlab.yml

solidnerd committed:
Upgrade GitLab CE to 12.2.3

solidnerd committed:
Update gitaly to 1.59.2

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
Update gitlab-workhorse to 8.8.1

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
Upgrade GitLab CE to 12.2.4

solidnerd committed:
release 12.2.4

solidnerd committed:
Upgrade GitLab CE to 12.2.5

solidnerd committed:
Update gitlab-pages to 1.7.2

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
release 12.2.5

tomislater committed:
Change VERSION to 12.2.5

solidnerd committed:
Upgrade GitLab CE to 12.3.0

solidnerd committed:
Update gitlab-shell to 10.0.0

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
Update gitlab-workhorse to 8.10.0

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
Update gitlab-pages to 1.9.0

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
Update gitaly to 1.65.0

Signed-off-by: solidnerd <niclas@mietz.io>

JGrenier committed:
Fix Auth0 config (#1964)

Fix Auth0 configuration and Add OAUTH_AUTH0_SCOPE

sue445 committed:
Add prepared_statements to database.yml (#1994)

* Add prepared_statements  to database.yml

Co-Authored-By: Niclas Mietz <github@mietz.io>

pbe-axelor committed:
Set Sidekiq default log format to json

This follow gitlab default value.
See https://gitlab.com/gitlab-org/gitlab-foss/merge_requests/32842

Signed-off-by: Pierre Belloy <p.belloy@axelor.com>

solidnerd committed:
Fix Auth0 Scope config parsing (#2007)

solidnerd committed:
Increase waiting to 180 seconds for circle ci

solidnerd committed:
release 12.3.0

solidnerd committed:
Upgrade GitLab CE to 12.3.1

solidnerd committed:
release 12.3.1

pbe-axelor committed:
Use updated gitlab-ce repository name `gitlab-foss`

Gitlab is moving to a single codebase for GitLab Community and Enterprise editions.
All code and development is moved into the gitlab-ee repository.
The code of the gitlab-ce repository remains as is, and won't include any proprietary code.
The `gitlab-ce` repository is renamed to `gitlab-foss`

See https://about.gitlab.com/2019/08/23/a-single-codebase-for-gitlab-community-and-enterprise-edition/

Signed-off-by: Pierre Belloy <p.belloy@axelor.com>

hroemer committed:
Documentation update on Docker config and secrets file env and example fix (#1980)

* fixed typos in referenced config and secret files

* updated documentation on external configs and secrets and added note on hard-coded config keys

* updated documentation link to sample config

* updated documentation link to sample config

* updated documentation links to config keys

* updated documentation links to config keys

* updated documentation links to config keys

pbe-axelor committed:
Add S/MIME Email Signing (#2009)

* Add S/MIME Email Signing

See https://docs.gitlab.com/ce/administration/smime_signing_email.html

Signed-off-by: Pierre Belloy <p.belloy@axelor.com>

* Remove s/mime email configs if not enabled

Signed-off-by: Pierre Belloy <p.belloy@axelor.com>

Commit:
Fix GITLAB_PAGES_ACCESS_REDIRECT_URI documentation

solidnerd committed:
Upgrade GitLab CE to 12.3.2

solidnerd committed:
Update gitaly to 1.65.1

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
release 12.3.2

phenomax committed:
upgrade CE to v12.3.3

solidnerd committed:
fix: increase circle no_output_timeout to 60 minutes

solidnerd committed:
release 12.3.3

solidnerd committed:
Upgrade GitLab CE to 12.3.4

solidnerd committed:
release 12.3.4

solidnerd committed:
Use docker volumes instead of bind mounts

abeltay committed:
Upgrade GitLab CE to 12.3.5

solidnerd committed:
release 12.3.5

splattael committed:
Remove deprecated "ssl" directives from nginx

Before this commit running "nginx -c -f /etc/nginx/nginx.conf" warned:

    nginx: [warn] the "ssl" directive is deprecated, use the "listen ... ssl" directive instead in /etc/nginx/sites-enabled/gitlab:60

splattael committed:
Remove remaining references to MySQL

MySQL support was removed in https://github.com/sameersbn/docker-gitlab/pull/1961

splattael committed:
Hide unused envvar DB_COLLATION

MySQL support was removed in https://github.com/sameersbn/docker-gitlab/pull/1961
and PostgreSQL doesn't support it anyway.

splattael committed:
Enforce `CI_REGISTRY_PASSWORD` dropping `CI_REGISTRY_PASS`

solidnerd committed:
Upgrade GitLab CE to 12.4.0

solidnerd committed:
Update gitlab-shell to 10.2.0

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
Update gitaly to 1.67.0

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
Update gitlab-workhorse to 8.14.0

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
Update gitlab-pages to 1.11.0

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
update base-image to ubuntu:bionic-20191010

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
Update golang to 1.12.12

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
release 12.4.0

solidnerd committed:
Upgrade GitLab CE to 12.4.1

solidnerd committed:
release 12.4.1

Commit:
gitlab: upgrade CE to v12.4.2

solidnerd committed:
release 12.4.2

pbe-axelor committed:
Upgrade GitLab CE to 12.4.3

solidnerd committed:
release 12.4.3

solidnerd committed:
Upgrade GitLab CE to 12.5.0

solidnerd committed:
Update gitaly to 1.72.0

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
Update gitlab-pages to 1.12.0

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
release 12.5.0

solidnerd committed:
Upgrade GitLab CE to 12.5.1

solidnerd committed:
Update gitlab-workhorse to 8.14.1

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
Update gitaly to 1.72.1

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
release 12.5.1

solidnerd committed:
Upgrade GitLab CE to 12.5.2

solidnerd committed:
release 12.5.2

solidnerd committed:
Upgrade GitLab CE to 12.5.3

Commit:
gitlab: upgrade CE to v12.5.4

Commit:
Update golang to 1.12.14

solidnerd committed:
release 12.5.3

solidnerd committed:
release 12.5.4

solidnerd committed:
Upgrade GitLab CE to 12.5.5

solidnerd committed:
release 12.5.5

solidnerd committed:
Upgrade GitLab CE to 12.6.0

solidnerd committed:
Update gitaly to 1.78.0

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
Update gitlab-shell to 10.3.0

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
Update gitlab-workhorse to 8.18.0

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
Update gitaly to 1.78.0

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
Update baseimage bionic to 20191202

sachilles committed:
fix(gitlab-shell): let gitlab-shell compile again

eiabea committed:
Merge branch 'master' into master

Commit:
Upgrade GitLab CE to 12.5.6

sachilles committed:
Update gitlab-workhorse to 8.19.0

sachilles committed:
Update gitlab-pages to 1.14.0

sachilles committed:
Update gitaly to 1.83.0

sachilles committed:
Update golang to 1.13.6

sachilles committed:
Upgrade GitLab CE to 12.7.0

sachilles committed:
Update gitlab-shell to 11.0.0

sachilles committed:
Update Changelog.md

solidnerd committed:
release 12.5.6

solidnerd committed:
Upgrade GitLab CE to 12.5.7

solidnerd committed:
release 12.5.7

solidnerd committed:
release 12.6.0

solidnerd committed:
Upgrade GitLab CE to 12.6.1

solidnerd committed:
release 12.6.1

solidnerd committed:
Upgrade GitLab CE to 12.6.2

solidnerd committed:
release 12.6.2

solidnerd committed:
Upgrade GitLab CE to 12.6.3

solidnerd committed:
Update gitaly to 1.77.1

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
release 12.6.3

solidnerd committed:
Upgrade GitLab CE to 12.6.4

solidnerd committed:
release 12.6.4

sachilles committed:
Upgrade GitLab CE to 12.7.2

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
release 12.7.0

sachilles committed:
Update gitlab-workhorse to 8.20.0

sachilles committed:
Update gitlab-pages to 1.15.0

sachilles committed:
Update gitaly to 1.85.0

sachilles committed:
Update golang to 1.13.7

sachilles committed:
Upgrade GitLab CE to 12.7.4

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
release 12.7.2

solidnerd committed:
release 12.7.4

solidnerd committed:
Upgrade GitLab CE to 12.7.5

solidnerd committed:
Update gitlab-pages to 1.12.0

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
Update gitaly to 1.83.0

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
release 12.7.5

solidnerd committed:
Upgrade GitLab CE to 12.7.6

solidnerd committed:
release 12.7.6

Commit:
fix: test image curl uses ipv4 only

* Despite the command running with --retry-connrefused, connections with ipv6 are not retried because it returns EADDRNOTAVAIL instead of ECONNREFUSED

Reference: https://github.com/appropriate/docker-curl/issues/5

tomastzn committed:
Include note about PostgreSQL version 10

GitLab works fine with version PostgreSQL release 11.x,  however backups will fail because the bundled tools, `postgresql-client-10` and `postgresql-contrib-10` are from release 10.x and they will not work with other releases.

KashifSaadat committed:
Allow modifying the sshd listen port

Commit:
Upgrade GitLab CE to 12.7.7

solidnerd committed:
release 12.7.7

BartJoris committed:
Add Generic OAuth2 provider (#2070)

* Add Generic OAuth2 provider

* Add docs to test Keycloak on a local environment

* Update docs for oauth2_generic

Commit:
Add logging for backup operations to file ${GITLAB_LOG_DIR}/gitlab/gitlab-backup.log

alessandrotrisolini committed:
Upgrade GitLab CE to 12.7.8 (#2106)

Co-authored-by: Alessandro Trisolini <alessandro@unumotors.com>

solidnerd committed:
release 12.7.8

solidnerd committed:
Upgrade to 12.8.0 (#2090)

* Upgrade GitLab CE to 12.8.0

* Update gitlab-workhorse to 8.21.0

Signed-off-by: solidnerd <niclas@mietz.io>

* Update gitlab-pages to 1.16.0

Signed-off-by: solidnerd <niclas@mietz.io>

* Update gitaly to 12.8.0

Signed-off-by: solidnerd <niclas@mietz.io>

* fix: gitaly-hooks missing symlink

Signed-off-by: solidnerd <niclas@mietz.io>

* chore(base-image): update base image to 20200219

* fix: unicorn worker limits

Signed-off-by: solidnerd <niclas@mietz.io>

* fix(ci): disable caching

Signed-off-by: solidnerd <niclas@mietz.io>

* feat: add patch support for gitlab during the installation

Signed-off-by: solidnerd <niclas@mietz.io>

* fix: ArgumentError: 'import/{{oauth2_generic_name}}' is not a supported controller name

Closes #2101

Signed-off-by: solidnerd <niclas@mietz.io>

* fix: add fix for upstream gitaly problems during a fresh install

Signed-off-by: solidnerd <niclas@mietz.io>

* chore: update changelog.md

solidnerd committed:
release 12.8.0

solidnerd committed:
Upgrade GitLab CE to 12.8.1 (#2108)

solidnerd committed:
release 12.8.1

solidnerd committed:
Upgrade GitLab CE to 12.8.2 (#2109)

solidnerd committed:
release 12.8.2

solidnerd committed:
Upgrade GitLab CE to 12.8.3 (#2110)

solidnerd committed:
release 12.8.3

solidnerd committed:
Upgrade GitLab CE to 12.8.4 (#2111)

solidnerd committed:
release 12.8.4

solidnerd committed:
Upgrade GitLab CE to 12.8.5 (#2112)

solidnerd committed:
release 12.8.5

solidnerd committed:
Upgrade GitLab CE to 12.8.6 (#2113)

solidnerd committed:
release 12.8.6

solidnerd committed:
Upgrade GitLab CE to 12.8.7 (#2114)

solidnerd committed:
release 12.8.7

solidnerd committed:
Upgrade to 12.8.8 (#2115)

* Upgrade GitLab CE to 12.8.8

* Update gitlab-workhorse to 8.21.1

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
release 12.8.8

solidnerd committed:
Upgrade to 12.9.0 (#2117)

* Upgrade GitLab CE to 12.9.0

* Update gitlab-shell to 12.0.0

Signed-off-by: solidnerd <niclas@mietz.io>

* Update gitlab-workhorse to 8.25.0

Signed-off-by: solidnerd <niclas@mietz.io>

* Update gitlab-pages to 1.17.0

Signed-off-by: solidnerd <niclas@mietz.io>

* Update gitaly to 12.9.0-rc42

Signed-off-by: solidnerd <niclas@mietz.io>

* feat: replace unicorn with puma as webserver

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
release 12.9.0

solidnerd committed:
Upgrade to 12.9.1 (#2118)

* Upgrade GitLab CE to 12.9.1

* Update gitlab-workhorse to 8.25.1

Signed-off-by: solidnerd <niclas@mietz.io>

* Update gitaly to 12.9.1

Signed-off-by: solidnerd <niclas@mietz.io>

solidnerd committed:
release 12.9.1

solidnerd committed:
Upgrade GitLab CE to 12.9.2 (#2119)

solidnerd committed:
release 12.9.2

sue445 committed:
[Doc] Migrate doc for unicorn parameters to puma (#2123)

unicorn is removed at #2117

sue445 committed:
Pass parameters of puma_worker_killer to puma config (#2124)

Commit:
allows generic name

Commit:
Upgrade gitlab to 12.9.4

sameersbn committed:
Merge pull request #2129 from drujd/12.9.4

Upgrade gitlab to 12.9.4

sameersbn committed:
readme: remove links to quay registry

The `assets:compile` rake task requests more memory than the quay.io
build machine can provide. As a result builds on quay.io have been
failing for a long time now.

sameersbn committed:
build: clean up build caches from `bundle` and `go` steps

sameersbn committed:
Merge pull request #2132 from sameersbn/remove-quay

readme: remove links to quay registry

sameersbn committed:
Merge pull request #2133 from sameersbn/clean-up

build: clean up build caches from `bundle` and `go` steps

sameersbn committed:
update ci pipeline to circleci 2.1

sameersbn committed:
dockerfile: move `ARG` lines to improve cache hits

sameersbn committed:
gitlab: updated to 12.9.5

sameersbn committed:
gitlab-shell: updated to 12.2.0

sameersbn committed:
gitaly: updated to 12.10.0

sameersbn committed:
gitaly: updated to 12.10.0

sameersbn committed:
update docs for release

sameersbn committed:
update docs for release

sameersbn committed:
Merge branch 'gitlab-12.9.5' into 'master'

Gitlab 12.9.5

See merge request docker-images/gitlab!3

sameersbn committed:
Merge pull request #2135 from sameersbn/gitlab-12.9.5

gitlab: release 12.9.5

sameersbn committed:
Merge pull request #2134 from sameersbn/ci-update

update ci pipeline to circleci 2.1

sameersbn committed:
Merge pull request #2126 from tom-dierckx/master

generic_oauth variable name for automatic authentication

sameersbn committed:
Merge pull request #2097 from elitvinov/gitlab-backup-logging

Add logging for backup operations

sameersbn committed:
Merge pull request #2093 from appvia/change-sshd-listen-port

Allow modifying the sshd listen port

sameersbn committed:
ci: specify correct build-args

sameersbn committed:
Merge pull request #2137 from sameersbn/ci-fix-build-args

ci: specify correct build-args

sameersbn committed:
Merge pull request #1925 from emon/oauth_saml_attribute_statements

fix gitlab_configure_oauth_saml_attribute_statements()

sameersbn committed:
remove "Fix for rebase in forks"

sameersbn committed:
makefile: add buildargs to build commands

sameersbn committed:
integrations: add stalebot

sameersbn committed:
Merge pull request #2138 from sameersbn/remove-temp-fixes

remove "Fix for rebase in forks"

sameersbn committed:
Merge pull request #2139 from sameersbn/stale-bot

integrations: add stalebot

sameersbn committed:
Merge pull request #1443 from kauden/patch-2

Accelerate map_uidgid

sameersbn committed:
Merge branch 'master' into master

sameersbn committed:
Merge pull request #1472 from eiabea/master

Fixed readme to match new backup file naming

achretien committed:
handle new gitaly logs correctly

sameersbn committed:
Merge pull request #2140 from achretien/update-gitaly-config

handle new gitaly logs correctly

sameersbn committed:
updated to ubuntu:bionic-20200403

sameersbn committed:
gitlab-workhorse: update to 8.30.1

sameersbn committed:
sync: upstream configs

sameersbn committed:
gitlab: upgrade to 12.10.4

solidnerd committed:
Upgrade GitLab CE to 12.10.6

solidnerd committed:
docs: update changelog for latest release

solidnerd committed:
fix: add GITLAB_HOME env for gitlab.yml substitution

This adds a missing environment variable that was required in
https://github.com/sameersbn/docker-gitlab/blob/master/assets/runtime/config/gitlabhq/gitlab.yml#L1100

Closes #2145

t7tran committed:
Fix accidental replacement if no backup is configured

If none of AWS or GCS backups are configured, sed command would accidentially replace the second "upload:" occurrence until the end of file.

solidnerd committed:
fix: duplicate remote directory key and aws encryption settings

chasemaier commented:
Something seems off here.

Initially:

* We need to get fully update with upstream repo (e.g. https://github.com/sameersbn/docker-gitlab/tags?after=13.0.0, they have tags/branches that are not in this repo).

To produce a new change we need to:

* Create a csats branch based on the best available branch/tag from upstream git (the version you want to go to is the base of your branch)
* Add commits to that to customize it (changes to readme, changes from ce => ee flavor globally, will need to be sure it's exhaustive) -- e.g. this diff https://github.com/sameersbn/docker-gitlab/compare/master...csats:docker-gitlab:csats-v12.2.1-ee

If you need an upstream (gitlab) patch that is unavailable in the upstream git repo, we can consider attempting that patch update, but I'm not sure if it's simply changing the version number in one place, we would need to be very sure there were not other parts that need attention. You may be able to discover from upstream what goes into a patch version bump -- I have never investigated..

We can PRs for review if you'd like, but we'll never actually merge anything here.

skumar-jnj commented:
Ah, yep.  That all makes sense.  The assumption I went on (rather, naive hope) was that a minor version bump would basically be a one-liner change in the Dockerfile.  However, this is amnesia on my part.  I've been bitten hard in the past by that assumption which I'll chalk it up to devops rustiness today.  Lemme see what I can do here and thanks for the feedback and helpful link.

skumar-jnj committed:
Sync with upstream 12.10.6, update pertient files for ee (*-foss =>
*-ee), update files to build without error, sync with gitaly, update readme
JATV-789

chasemaier commented:
I am looking at this diff: https://github.com/sameersbn/docker-gitlab/compare/12.10.6-1...csats:docker-gitlab:csats-v12.10.14-ee

I see at first glance `GITALY_SERVER_VERSION=12.10.6` in the Dockerfile is the "changed" version but remains at the `.6` patch release. I'd do a global search to see if there are any more "similar" ones elsewhere. What I don't know is if these two (or more) have to stay in sync in any way or if they are allowed to safely diverge. I'd consider either:

* Finding documentation that says what they even are and if the versions are compatible
* Possibly trying to bump the gitaly version (etc...) at the same time to keep it in sync? Looking how the upstream maintainer changes these values between releases may give signals about how they are "typically" treated.
